<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1243</width>
    <height>804</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Обработка изображений</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="res">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>10</y>
      <width>600</width>
      <height>560</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="src">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>600</width>
      <height>560</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QGroupBox" name="processing">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>610</y>
      <width>1221</width>
      <height>181</height>
     </rect>
    </property>
    <property name="title">
     <string>Обработка изображений</string>
    </property>
    <widget class="QPushButton" name="useResult">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Использовать
 результат</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="findEdges">
     <property name="geometry">
      <rect>
       <x>710</x>
       <y>30</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Выделение границ</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">editAct</string>
     </attribute>
    </widget>
    <widget class="QComboBox" name="noiseFilterMethod">
     <property name="geometry">
      <rect>
       <x>530</x>
       <y>80</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Медианный фильтр</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Фильтр Гаусса</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Фильтр &quot;Усреднение&quot;</string>
      </property>
     </item>
    </widget>
    <widget class="QRadioButton" name="noiseFilter">
     <property name="geometry">
      <rect>
       <x>530</x>
       <y>30</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Удаление шума</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">editAct</string>
     </attribute>
    </widget>
    <widget class="QComboBox" name="grayScaleMethod">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>80</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Average method</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Lightness method</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Luminosity method</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Photoshop, Gimp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ITU-R, BT.709</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Max из трёх каналов</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Min из трёх каналов</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>(0.2952, 0.5547, 0.148)</string>
      </property>
     </item>
    </widget>
    <widget class="QSlider" name="filterSize">
     <property name="geometry">
      <rect>
       <x>530</x>
       <y>130</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="maximumSize">
      <size>
       <width>160</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>7</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
     <property name="sliderPosition">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
     <property name="tickInterval">
      <number>0</number>
     </property>
    </widget>
    <widget class="QComboBox" name="findEdgesMethod">
     <property name="geometry">
      <rect>
       <x>710</x>
       <y>80</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Оператор Робертса</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Оператор Прюитт</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Оператор Собеля</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="ApplyButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Применить</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="isAllChannels">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>130</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>По каждому каналу</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="threshold">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>80</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>128</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="grayScale">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>30</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Перевод в оттенки 
серого</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">editAct</string>
     </attribute>
    </widget>
    <widget class="QLabel" name="filterSizeInfo">
     <property name="geometry">
      <rect>
       <x>650</x>
       <y>130</y>
       <width>21</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>3</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="Binarization">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>30</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Бинаризация</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">editAct</string>
     </attribute>
    </widget>
    <widget class="QRadioButton" name="morphFilter">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>30</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Морфологическая
фильтрация</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">editAct</string>
     </attribute>
    </widget>
    <widget class="QRadioButton" name="filter">
     <property name="geometry">
      <rect>
       <x>1040</x>
       <y>30</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Фильтры общего вида </string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">editAct</string>
     </attribute>
    </widget>
    <widget class="QComboBox" name="BinarizationMethod">
     <property name="geometry">
      <rect>
       <x>320</x>
       <y>80</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Пороговая, вручную</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Метод Оцу</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="morphFilterMethod">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>81</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Наращивание</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Эрозия</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Открытие</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Закрытие</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Top Hat</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Black Hat</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Морф. выд. границ</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="filterMethod">
     <property name="geometry">
      <rect>
       <x>1040</x>
       <y>80</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Повышение резкости</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Барельеф</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Инверсия</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="openButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Открыть</string>
     </property>
    </widget>
    <widget class="QSlider" name="morphSize">
     <property name="geometry">
      <rect>
       <x>880</x>
       <y>130</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>7</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
     <property name="sliderPosition">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="morphSizeInfo">
     <property name="geometry">
      <rect>
       <x>970</x>
       <y>130</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>3</string>
     </property>
    </widget>
   </widget>
  </widget>
  <action name="open">
   <property name="text">
    <string>Открыть</string>
   </property>
  </action>
  <action name="exit">
   <property name="text">
    <string>Выход</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="editAct"/>
 </buttongroups>
</ui>
